### PR TITLE
add appdata to the project complexmodel

### PR DESCRIPTION
**Pull request by knoxsp**
This is a supporting PR to this one on hydra base:
https://github.com/hydraplatform/hydra-base/pull/198

### CODE REVIEW

File: hydra_server/server/complexmodels.py [hunk 1]
```diff
@@ -1480,6 +1480,7 @@ class Project(Resource):
         ('status',      Unicode(default='A', pattern="[AX]")),
         ('cr_date',     Unicode(default=None)),
         ('created_by',  Integer(default=None)),
+        ('appdata',     AnyDict(min_occurs=0, max_occurs=1, default=None)),
```
Comment (gbasolu) [inline hydra_server/server/complexmodels.py:1483]: Did you prefer anydict over json to be more flexible?

This field should also appear into the doc.

Comment (knoxsp) [reply hydra_server/server/complexmodels.py:1483]: This is just for consistency with other paramters in the file

Comment (knoxsp) [reply hydra_server/server/complexmodels.py:1484]: Added to docstring

File: hydra_server/server/complexmodels.py [hunk 2]
```diff
@@ -1496,6 +1498,16 @@ def __init__(self, parent=None):
         self.status      = parent.status
         self.cr_date     = str(parent.cr_date)
         self.created_by  = parent.created_by
+
+        #this is a failsafe to ensure that the 'appdata' column
+        #is processed correctly, as it is a JSON column type which may
+        #not work with with serialising data.
```
Comment (pmslavin) [inline hydra_server/server/complexmodels.py:1504]:
```suggestion
        #not work with serialising data.
```

File: hydra_server/server/complexmodels.py [hunk 3]
```diff
@@ -1496,6 +1498,16 @@ def __init__(self, parent=None):
...
+        appdata = {}
+        if parent.appdata is not None:
+            for k, v in parent.appdata.items():
+                appdata[k] = v
+        self.appdata = appdata
```
Comment (pmslavin) [inline hydra_server/server/complexmodels.py:1509]: This creates a copy of `parent.appdata` and then sets `self.appdata` to it, so is equivalent to...

```python
    if parent.appdata is not None:
        self.appdata = parent.appdata.copy()
```

Is `parent` guaranteed to have an `appdata` attr here?

**Review by pmslavin — COMMENTED** [review]
Comment text pedantic change and dict copy suggestion.  Seems fine otherwise with doc change from @gbasolu.

**Pull request merged**